### PR TITLE
[MISC] Adds new concept template_specialisation_of.

### PR DESCRIPTION
**Pull request by Irallia**
Resolves partly https://github.com/seqan/product_backlog/issues/259

### CODE REVIEW

**Comment by marehr** [comment]
Did you talk with @smehringer about this? Or did you pick this yourself?

**Comment by Irallia** [comment]
> 
> 
> Did you talk with @smehringer about this? Or did you pick this yourself?

She gave me this issue.

**Comment by marehr** [comment]
I guess this is blocked by #2265. We should first do the move of `{core/type_traits/ => core/detail/}template_inspection.hpp`

See https://gist.github.com/rrahn/f6b09ad67bebd14bd9c58f77c53e0450

**Review by marehr — COMMENTED** [review]
Postponed until the header template_inspection.hpp was moved.

**Review by marehr — APPROVED** [review]
thank you

File: include/seqan3/core/detail/template_inspection.hpp
Comment (smehringer) [inline include/seqan3/core/detail/template_inspection.hpp:253]:
```suggestion
 * \tparam mytype           The guery type.
```

? 
Source type somehow threw me off. Maybe query isn't the best alternative thought. `The type to check` ?
Source type sounds like I'm actually doing something with it or taking something from it to put somewhere else..

File: test/snippet/core/detail/template_inspection_usage_3.cpp
Comment (smehringer) [inline test/snippet/core/detail/template_inspection_usage_3.cpp:9]:
```suggestion
    if constexpr (seqan3::detail::template_specialisation_of<my_type, std::vector>) // Note: std::vector has no <> !
```

**Review by smehringer — APPROVED** [review]
Just to tiny documentation things. After including them this can be merged.

**Pull request merged**
